### ISSUE TITLE
Textures appear blurry when map imported into Godot

### ISSUE THREAD

**Issue by gulagkulak** [QodotPlugin/Qodot#83]
Using Godot 4.1.1 mono with latest Qodot from the asset library. Set up my workspace according to this (slightly outdated) tutorial: https://www.youtube.com/watch?v=dVagDDRb2jQ 

In the tutorial (around 7:10) she points out that by default textures are imported as compressed, which results in a blurry look, so she re-imports the textures without compression and gets a crisp pixel-art look right inside Godot.

I have previously had a blurry texture problem when importing Blender models and re-importing the textures as lossless fixed that. Tried the same thing with Qodot and... textures are still blurry.

This is what I see in Trenchbroom:

![image](https://github.com/QodotPlugin/Qodot/assets/144597589/f46c6476-dc01-4ce2-ad18-778a1a11fcbf)

And this blurry mess is what appears in Godot:

![image](https://github.com/QodotPlugin/Qodot/assets/144597589/f3d04499-855a-497b-bd8e-ba9481b33904)

That tiny table is a Blender model with pixel-perfect crisp textures. The Qodot map node, however, has blurry textures and no amount of re-importing textures with lossless compression (as suggested in the tutorial video) does anything. In the video she also unchecks a mysterious "filter" setting, which is not present in Godot 4.1.1.

I can, however, re-import the textures with compression to make them look even worse.

Setting the default texture filter to "nearest" in Godot settings didn't help.

How do I unblur textures in Qodot?

**Comment by DeficientNull** [comment]
Hello, in the new version of Qodot you need to change the default material definition automatically created by the plugin

Access the Child Nodes inside of QodotMap (you're looking for the meshes) in the properies tab, you can open the mesh and find a list of surfaces, when you open them you'll see that surface's material, there you can open the material and  go to the sampling tab to click the "Nearest" option

Hope this helps

**Comment by gulagkulak** [comment]
> Hello, in the new version of Qodot you need to change the default material definition automatically created by the plugin
> 
> Access the Child Nodes inside of QodotMap (you're looking for the meshes) in the properies tab, you can open the mesh and find a list of surfaces, when you open them you'll see that surface's material, there you can open the material and go to the sampling tab to click the "Nearest" option
> 
> Hope this helps

Well, that fixed the issue. Then I went back to Trenchbroom, made a small change to the map, hit "full build" in Godot and now everything is blurry again...

Is there a way to either automate this stuff or do I need to create custom materials for each texture I use to get proper pixel-perfect textures every time?

**Comment by DeficientNull** [comment]
i believe once you've changed the material you can drag it into the textures folder and trenchbroom will automatically assign it properly, but idk yet still figuring it out

**Comment by gulagkulak** [comment]
> i believe once you've changed the material you can drag it into the textures folder and trenchbroom will automatically assign it properly, but idk yet still figuring it out

Yup, you can drag the material to the same folder where your texture png is located and if you give it the same name as the texture png (but with a .tres extention instead of .png) then map updating works properly and doesn't blur things.

I wish there was a way to create the .tres files automatically when doing a full build.

**Comment by DeficientNull** [comment]
Welp, nice to know it worked for you at least

**Comment by gulagkulak** [comment]
Hey, I figured it out! Select the QodotMap node and in the inspector there is a "Default Material" property. If you change sampling to "nearest" on the default material, you'll get pixel-perfect materials every time!

**Comment by DeficientNull** [comment]
Oh, well ok then.
